### PR TITLE
a more configureable resnet3d implementation

### PR DESCRIPTION
**Pull request by avihu111**
controls conv channels, strides and stem kernel size

### CODE REVIEW

File: fuse/dl/models/backbones/backbone_resnet_3d.py [hunk 1]
```diff
@@ -17,78 +17,163 @@
...
-    def features(self, x: Tensor) -> Any:
-        """
-        Extract spatial features - given a 3D tensor
-        :param x: Input tensor - shape: [batch_size, channels, z, y, x]
-        :return: spatial features - shape [batch_size, n_features, z', y', x']
-        """
-        x = self.stem(x)
+        return nn.Sequential(*layers)
 
+    def features(self, x: Tensor) -> Tensor:
```
Comment (mosheraboh) [inline fuse/dl/models/backbones/backbone_resnet_3d.py:167]: Can you please add back the comments?

File: fuse/dl/models/backbones/backbone_resnet_3d.py [hunk 2]
```diff
@@ -17,78 +17,163 @@
...
+        *,
+        in_channels: int = 3,
+        pool: bool = True,
+        layers: List[int] = [2, 2, 2, 2],
+        first_channel_dim: int = 64,
+        first_stride: int = 1,
+        stem_kernel_size: Sequence[int] = (3, 7, 7),
+        stem_stride: Sequence[int] = (1, 2, 2),
+        pretrained: bool = False,
+        name: str = "r3d_18",
```
Comment (mosheraboh) [inline fuse/dl/models/backbones/backbone_resnet_3d.py:112]: Can you document the important arguments like name and specify what the options are?

**Review by mosheraboh — COMMENTED** [review]
Looks good!
Two comments inline.

File: fuse/dl/models/backbones/backbone_resnet_3d.py [hunk 3]
```diff
@@ -17,78 +17,163 @@
...
-        }[name]
-        # init original model
-        super().__init__(**init_parameters)
-
-        # load pretrained parameters if required
+    def __init__(
+        self,
+        *,
+        in_channels: int = 3,
+        pool: bool = True,
```
[the remark this answers is not in the document]
Comment (avihu111) [reply fuse/dl/models/backbones/backbone_resnet_3d.py:105]: I think the default should be "True", as this is the common usage (change the test instead).
Other consideration is backward compatibility.

File: fuse/dl/models/backbones/backbone_resnet_3d.py (hunk 1 above)
Comment (avihu111) [reply fuse/dl/models/backbones/backbone_resnet_3d.py:167]: adding and making them more accurate.

File: fuse/dl/models/backbones/backbone_resnet_3d.py (hunk 2 above)
Comment (avihu111) [reply fuse/dl/models/backbones/backbone_resnet_3d.py:112]: done

**Review by mosheraboh — APPROVED** [review]

**Pull request merged**